pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -141,14 +168,21 @@ pub mod pallet {
 			domain: Domain,
 			hook_address: [u8; 20],
 		},
+
+		/// Message recovery was executed.
+		MessageRecoveryExecuted {
+			proof: Proof,
+			router_id: T::RouterId,
+		},
 	}
 
-	/// Storage for domain routers.
+	/// Storage for routers.
 	///
 	/// This can only be set by an admin.
 	#[pallet::storage]
-	#[pallet::getter(fn domain_routers)]
-	pub type DomainRouters<T: Config> = StorageMap<_, Blake2_128Concat, Domain, T::Router>;
+	#[pallet::getter(fn routers)]
+	pub type Routers<T: Config> =
+		StorageValue<_, BoundedVec<T::RouterId, T::MaxRouterCount>, OptionQuery>;
```

Comment:
NIT. Maybe `ValueQuery` does the trick? if it's not configured is the same as configured with an empty router list.

Reply:
I had `ValueQuery` for this and the session ID but somehow it felt a bit weird to deal with empty vectors in this case + not having a state where no routers are set. Maybe I'm overthinking this and paranoia got to me...

Reply:
Done separately [here](https://github.com/centrifuge/centrifuge-chain/pull/1965).